src/app/core/public/api/impl/Circuit.ts

```diff
@@ -155,7 +159,23 @@ export abstract class CircuitImpl implements Circuit {
         //  For now, ignore the `space`, and ignore any non-Component
         //   objects that are selected
         //  From these components, average their positions
-        throw new Error("Method not implemented.");
+        const allComponents = this.selections.get();
+
+        // Case: no components are selected
+        if (allComponents.length === 0) {
+            return V(0,0)
+        }
+
+        // Case: One or more components are selected
+        let comps =  allComponents.map((id) => this.getComponent(id))
+                    .filter((comp) => (comp !== undefined)) as Component[];
+        let sumPositions = comps.reduce((vec1, vec2) => V(vec1.x, vec1.y).add(vec2.pos.x, vec2?.pos.y), V(0,0));
+
+        // Calculate average position
+        let xPosition = sumPositions.x / allComponents.length;
+        let yPosition = sumPositions.y / allComponents.length;
+
+        return V(xPosition, yPosition);
```

Comment:
And again `Vector` provides methods so you don't have to do things individually

simply `return sumPosition.scale(1 / allComponents.length);`